src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -0,0 +1,229 @@
+import { html, LitElement } from "@polymer/lit-element";
+import { repeat } from "lit-html/directives/repeat";
+import "@polymer/paper-checkbox/paper-checkbox.js";
+import "@polymer/paper-icon-button/paper-icon-button.js";
+import "@polymer/paper-input/paper-input.js";
+import "@polymer/paper-item/paper-icon-item.js";
+import "@polymer/paper-item/paper-item-body.js";
+
+import "../../../components/ha-card.js";
+
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../types.js";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import {
+  clearItems,
+  fetchItems,
+  addItem,
+  completeItem,
+  saveEdit,
+} from "../../../data/shopping_list";
+import { TemplateResult } from "lit-html";
+
+interface Config extends LovelaceConfig {
+  title?: string;
+}
+
+export interface ShoppingListItem {
+  id: number;
+  name: string;
+  complete: boolean;
+}
+
+class HuiShoppingListCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  protected config?: Config;
+  private items?: ShoppingListItem[];
+  private _unsubEvents?: () => Promise<void>;
+
+  static get properties() {
+    return {
+      config: {},
+      items: {},
+    };
+  }
+
+  public getCardSize(): number {
+    return (
+      (this.config ? (this.config.title ? 1 : 0) : 0) +
+      (this.items ? this.items.length : 3)
+    );
+  }
+
+  public setConfig(config: Config): void {
+    this.config = config;
+    this.items = [];
+    this._fetchData();
+  }
+
+  public connectedCallback() {
+    if (this.hass) {
+      super.connectedCallback();
+      this.hass.connection
+        .subscribeEvents(this._fetchData, "shopping_list_updated")
+        .then(function(this: HuiShoppingListCard, unsub: () => Promise<void>) {
+          this._unsubEvents = unsub;
+        });
+      this._fetchData();
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+
+    if (this._unsubEvents) {
+      this._unsubEvents();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!this.config || !this.hass) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this.config.title}">
+        <div class="addRow">
+          <paper-input
+            id='addBox'
+            placeholder='${this.localize("ui.panel.shopping-list.add_item")}'
```

Comment:
The reason this doesn't work, is because the shopping list panel resources have not been loaded. It's all splitted up to minimize the amount of data our user has to load on startup. You should define new strings just for lovelace.

Here shopping list is defined: https://github.com/home-assistant/home-assistant-polymer/blob/master/src/translations/en.json#L835

Just add a new key as a sibling for lovelace. Under that make a new one card and then card type and then "add_item". Just fill in English, it will be automatically uploaded to the translation platform and our translator community will do the rest.